packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,449 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title The Nouns art storage contract
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.6;
+
+import { INounsArt } from './interfaces/INounsArt.sol';
+import { Inflate } from './libs/Inflate.sol';
+import { SSTORE2 } from './libs/SSTORE2.sol';
+
+contract NounsArt is INounsArt {
+    /// @notice Current Nouns Descriptor address
+    address public override descriptor;
+
+    /// @notice Pending Nouns Descriptor address
+    address public override pendingDescriptor;
+
+    /// @notice Noun Backgrounds (Hex Colors)
+    string[] public override backgrounds;
+
+    /// @notice Noun Color Palettes (Index => Hex Colors, stored as a contract using SSTORE2)
+    mapping(uint8 => address) public palettesPointers;
+
+    /// @notice Noun Bodies Trait
+    Trait public bodiesTrait;
+
+    /// @notice Noun Accessories Trait
+    Trait public accessoriesTrait;
+
+    /// @notice Noun Heads Trait
+    Trait public headsTrait;
+
+    /// @notice Noun Glasses Trait
+    Trait public glassesTrait;
+
+    /**
+     * @notice Require that the sender is the descriptor.
+     */
+    modifier onlyDescriptor() {
+        if (msg.sender != descriptor) {
+            revert SenderIsNotDescriptor();
+        }
+        _;
+    }
+
+    constructor(address _descriptor) {
+        descriptor = _descriptor;
+    }
+
+    /**
+     * @notice Set the pending descriptor, which can be confirmed
+     * by calling `confirmDescriptor`.
+     * @dev This function can only be called by the current descriptor.
+     */
+    function setDescriptor(address _pendingDescriptor) external override onlyDescriptor {
+        pendingDescriptor = _pendingDescriptor;
+    }
+
+    /**
+     * @notice Confirm the pending descriptor.
+     * @dev This function can only be called by the pending descriptor.
+     */
+    function confirmDescriptor() external override {
```

Comment:
The descriptor does not expose functions to call these

Reply:
👍 good catch.
will add a function in the descriptor, and reduce it to a one step.

Reply:
thank you, fixed!